api/src/main/java/org/apache/iceberg/FileScanTask.java

```diff
@@ -24,7 +24,7 @@
 /**
  * A scan task over a range of a single file.
  */
-public interface FileScanTask extends ScanTask {
+public interface FileScanTask extends ScanTask, Comparable<FileScanTask> {
```

Comment:
Not sure if this change to break the api compatibility.

Reply:
It shouldn't break compatibility, but I'd prefer not to add it to the interface. First, it isn't necessary to use this because we can use a separate `Comparator<FileScanTask>` that can be a private implementation. That makes more sense to me because this is an arbitrary ordering of tasks.

Second, while I can't really think of a case where someone would do something strange, this does make it possible. For example, if an implementation chose to override this with `return 0` then the source would have non-deterministic behavior. If we can avoid a larger API by using `Comparator` then I think that's a good idea.

Reply:
Sure, I agree, let me change to use `Comparator`.